### ISSUE TITLE
a question on pgnvjs.js

### ISSUE THREAD

**Issue by cgencer** [mliebelt/PgnViewerJS#101]
Hi again,
I have a question on adding moves for branches. I am mimicking a few parts of your drag/snap codes within my code (for adding custom behaviours). I use this code:
```
		var thePGN = theBoard.getPgn();
		if(currentMoveNotation !== ''){
			console.log(currentMoveNotation+'...'+themno);
			var curMove = thePGN.addMove(currentMoveNotation, themno);
			var pgnMove = thePGN.getMove(curMove);
			themno = pgnMove.moveNumber;
			console.dir(pgnMove);
			chess.load(pgnMove.fen);
	        cbUI.position(pgnMove.fen);
//			console.dir(pgnMove);

			$('#gamecode').text(thePGN.write_pgn());

```
This works fine for the first move in a new branch, but whenever I add a second move to this branch right afterwards, it gives error. 
TypeError: pgn_move is null (Line19612)
which points to addMove and then existing_move afterwards.

**themno** is the move no obtained from the 'to-be'branching' and existing move (e.g. **.moveNumber**). But as the second move added has no moveNumber attached to it, I cant add it to the branch. With removing  themno= line it adds many branches (with only one move) to the main line serially.( e.g. _xxx (n. yyy) (n. zzz)_ ) and so on. So I try to find a way to add moves sequentially to a specific main which I have the moveNumber for. If you have some minutes, I'd be glad to hear from you what is the required steps I need to take for adding moves into a branch. Vielen Dank für Ihre Zeit! :)

**Comment by mliebelt** [comment]
I checked the code (as good as possible from what I have seen), and I don't see a problem. The notation `currentMove` is misleading. In my case (playing on a visual board), I have all the time only one position which is on the board shown, and set in the `chess` object inside. The `currentMove` is the number to the array of moves held by the pgn object.

So I have no recommendation for what you want to do, just a tip:

* Use the original files and learn how to use them.
* All my examples (internally) use a library named `dev.js` (for development). This library is build by using  `grunt dev` from the command line.
* The examples then reference `pgn.js` and `pgnv.js` in source, this makes the debugging much easier.

It would help if I would understand what you want to do. So I am able to use the `pgnEdit` function to have a board to play new games (including variations). So the library is able to add variations without a problem.

I have some Jasmine tests that correspond to what you want to do. Have a look at `PgnViewerJS/test/spec/pgnSpec.js`, there the lines from 613. Try to create first a test case, and understand how `addMove` works. 

Give me some hints what you want to do, by telling me the whole scenario, and perhaps I am able to help then. And I will try to come up with a short description how to install the development version of PgnViewerJS, and to use that in your own code then.

**Comment by cgencer** [comment]
thanks a lot for taking the time, really i appreciate your help. kudos!
after examining deeply, i dropped my faulty approach and started from scratch with restyling the design for my purposes a bit. it went quite ok after your suggestions. i have now a few thoughts, maybe you can use the one or other...

- chessboard.js has an option to put pieces onto the board manually and remove them. when building problems / puzzles, it could be quite handy to remove some pieces from the table.
- it could be quite handy, if the pgnEdit function returns a real reference to access some methods. especially for reloading a pgn/fen, or programmatically advance/rewind the game. currently it returns undefined, so it strangled me to use a var container just to call the pgnEdit directly.
- do you plan to add painting functions for the earlier spoken arrow-drawing & field-coloring? if not, i'm gonna do these for myself for now. drawings are quite easy actually, but the harder part is that i'd had to insert these special comments into the generated pgn. 
- is 'en passant' movement supported? a few times while testing (of course with the vaniila examples you provided) it wouldn't remove the passed pawn from the table, but acted as if it's not present. i will try to create a failing case for it after doing a few more checks.
i already implemented an easy backend-saving & loading method using pouchdb, which also sync's itself easily. maybe i can later on write an addon-script for this. but i first plan to vue.js the whole thing... ;)

best!

**Comment by mliebelt** [comment]
Fixed #91 (drawing arrows and circles) as well, will close that question, because there will no further work on it.
